test/unit/errors/error-collector.test.js

```diff
@@ -2370,3 +2370,76 @@ test('When using the async listener', (t) => {
     })
   }
 })
+
+tap.test('_processErrors', (t) => {
+  let transaction = null
+  let errorCollector = null
+
+  t.beforeEach(() => {
+    t.context.agent = helper.loadMockedAgent({
+      attributes: {
+        enabled: true
+      }
+    })
+
+    transaction = new Transaction(t.context.agent)
+    transaction.url = '/'
+    errorCollector = t.context.agent.errors
```

Comment:
We want them all scoped the same way as the agent:
```suggestion
    const transaction = new Transaction(t.context.agent)
    transaction.url = '/'
    t.context.transaction = transaction
    
    t.context.errorCollector = t.context.agent.errors
```